src-opam/dockerfile_opam.ml

```diff
@@ -490,7 +494,7 @@ let all_ocaml_compilers hub_id arch distro =
     @@ env [ ("OPAMYES", "1") ]
     @@
     match os_family with
-    | `Linux | `Cygwin -> cmd "bash"
+    | `Linux | `Cygwin | `Macos -> cmd "bash" (* TODO Should we use zsh since that is the default now in MacOS? *)
```

Comment:
I don't think it would change much given this setting is only used for the dockerfile itself not really the underlying setup (which can't really be tested in CI)